src/meta/src/hummock/compaction_schedule_policy.rs

```diff
@@ -0,0 +1,722 @@
+                    if visited.contains(&compactor.context_id()) {
+                        return None;
+                    }
+                    if hummock_manager
+                        .get_assigned_tasks_number(compactor.context_id())
+                        .await
+                        < compactor.max_concurrent_task_number()
+                    {
+                        return Some(compactor);
+                    }
+                    visited.insert(compactor.context_id());
+                }
+            }
+        }
+    }
+
+    fn next_compactor(&mut self, _compact_task: Option<&CompactTask>) -> Option<Arc<Compactor>> {
+        if self.compactors.is_empty() {
+            return None;
+        }
+        let compactor_index = self.next_compactor % self.compactors.len();
+        let compactor = self.compactors[compactor_index].clone();
+        self.next_compactor += 1;
+        Some(compactor)
+    }
+
+    fn add_compactor(
+        &mut self,
+        context_id: HummockContextId,
+        max_concurrent_task_number: u64,
+    ) -> Receiver<MetaResult<SubscribeCompactTasksResponse>> {
+        let (tx, rx) = tokio::sync::mpsc::channel(STREAM_BUFFER_SIZE);
+        self.compactors.retain(|c| c.context_id() != context_id);
+        self.compactors.push(Arc::new(Compactor::new(
+            context_id,
+            tx,
+            max_concurrent_task_number,
+        )));
+        tracing::info!("Added compactor session {}", context_id);
+        rx
+    }
+
+    fn random_compactor(&mut self, _compact_task: Option<&CompactTask>) -> Option<Arc<Compactor>> {
+        if self.compactors.is_empty() {
+            return None;
+        }
+
+        let compactor_index = rand::thread_rng().gen::<usize>() % self.compactors.len();
+        let compactor = self.compactors[compactor_index].clone();
+        Some(compactor)
+    }
+
+    fn remove_compactor(&mut self, context_id: HummockContextId) {
+        self.compactors.retain(|c| c.context_id() != context_id);
+        tracing::info!("Removed compactor session {}", context_id);
+    }
+
+    fn report_compact_task(&mut self, _context_id: HummockContextId, _task: &CompactTask) {}
+}
+
+#[derive(Default)]
+pub struct LeastPendingBytesPolicy {
+    // These two data structures must be consistent.
+    // We use `(pending_bytes, context_id)` as the key to dedup compactor with the same pending
+    // bytes.
+    pending_bytes_to_compactor: BTreeMap<(usize, HummockContextId), Arc<Compactor>>,
+    compactor_to_pending_bytes: HashMap<HummockContextId, usize>,
+}
+
+impl LeastPendingBytesPolicy {
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    fn update_compactor(
+        &mut self,
+        context_id: HummockContextId,
+        old_bytes: usize,
+        compact_task: Option<&CompactTask>,
+        assign_task: bool,
+    ) -> Arc<Compactor> {
+        let task_pending_bytes = match compact_task {
```

Comment:
`pending_bytes` for per level, not only task_input, but also the out_put of the upper layer.
Maybe we can use `LevelHandler` to `get_pending_file_size` belonging to `compactor`. 

Shall we also need to `update_compactor` via `compact_task`  if we use `LevelHandler` ?